Core/include/Acts/Propagator/StepperExtensionList.hpp

```diff
@@ -63,25 +60,59 @@ struct StepperExtensionList : private detail::Extendable<extensions...> {
   template <typename propagator_state_t, typename stepper_t>
   bool validExtensionForStep(const propagator_state_t& state,
                              const stepper_t& stepper) {
-    std::array<int, nExtensions> bids;
-    // Ask all extensions for a boolean statement of their validity
-    impl::bid(tuple(), state, stepper, bids);
-    // Post-process the vector in an auctioneer
+    const auto bids = std::apply(
+        [&](const auto&... ext) {
+          return std::array<int, nExtensions>{ext.bid(state, stepper)...};
+        },
+        tuple());
+
     validExtensions = state.stepping.auctioneer(std::move(bids));
 
     return (std::find(validExtensions.begin(), validExtensions.end(), true) !=
             validExtensions.end());
   }
 
+  /// @brief This functions broadcasts the call for evaluating a generic k. It
+  /// collects all arguments and extensions, test their validity for the
+  /// evaluation and passes them forward for evaluation and returns a boolean as
+  /// indicator if the evaluation is valid.
+  template <typename propagator_state_t, typename stepper_t>
+  bool k(const propagator_state_t& state, const stepper_t& stepper,
+         Vector3& knew, const Vector3& bField, std::array<double, 4>& kQoP,
+         const int i, const double h = 0., const Vector3& kprev = Vector3{}) {
+    // TODO replace with integer-templated lambda with C++20
+    auto impl = [&, i, h](auto intType, auto& implRef) {
+      constexpr static int N = decltype(intType)::value;
```

Comment:
Does this have to be `static` here? Probably doesn't make a difference even if not.